Source/Core/DolphinQt/Debugger/CodeViewWidget.cpp

```diff
@@ -90,6 +158,11 @@ void CodeViewWidget::FontBasedSizing()
   Update();
 }
 
+u32 CodeViewWidget::AddressForRow(int row) const
+{
+  return m_address - ((rowCount() / 2) * 4) + row * 4;
```

Comment:
Thoughts on documenting those numbers (either as comment or `constexpr` if it makes sense to reuse them elsewhere)? Dividing by two and multiplying by four (is this the same four on both multiplies btw?) looks like magic and I cannot see a reason why either is necessary and/or yields the correct address here.

Reply:
That line was copied out of the row loop in `CodeViewWidget::Update()` so I'm not entirely sure what it's trying to do either. I think it's trying to align the address to be divisible by 4?

Reply:
Alright, I thought about it and ended up with:

~~~
  // m_address is defined as the center row of the table, so we have rowCount/2 instructions above
  // it; an instruction is 4 bytes long on GC/Wii so we increment 4 bytes per row
  const u32 rowZeroAddress = m_address - ((rowCount() / 2) * 4);
  return rowZeroAddress + row * 4;
~~~

Does that make sense to you?

Reply:
Seems better than before 👍